src/Compilers/Test/Utilities/CSharp/FunctionPointerUtilities.cs

```diff
@@ -128,6 +130,42 @@ static void verifyParameter(ParameterSymbol symbol, MethodSymbol containing)
             }
         }
 
+        public static void VerifyPublicFunctionPointerGetTypeInfo(SemanticModel model, SyntaxNode syntax, bool useConvertedType,
+            Action<ITypeSymbol> returnTypeVerifier,
+            params Action<ITypeSymbol>[] paramVerifiers)
+        {
+            var typeInfo = model.GetTypeInfo(syntax);
+            var typeSymbol = (IFunctionPointerTypeSymbol)(useConvertedType ? typeInfo.ConvertedType : typeInfo.Type)!;
+
+            if (!useConvertedType)
+            {
+                Assert.Equal(typeInfo.Type, typeInfo.ConvertedType, SymbolEqualityComparer.IncludeNullability);
+            }
+
+            CommonVerifyFunctionPointer((FunctionPointerTypeSymbol)((CSharp.Symbols.PublicModel.FunctionPointerTypeSymbol)typeSymbol).UnderlyingTypeSymbol);
```

Comment:
>(FunctionPointerTypeSymbol)((CSharp.Symbols.PublicModel.FunctionPointerTypeSymbol)typeSymbol).UnderlyingTypeSymbol [](start = 40, length = 114)

I think we have GetSymbol (or something like that) helpers to do this. Consider adding a helper for FunctionPointerTypeSymbol if necessary.

Reply:
Good to know, thanks.

---
In reply to: [428899939](https://github.com/dotnet/roslyn/pull/44436#discussion_r428899939) [](ancestors = 428899939)